daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/iterable/SExprIterable.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.speedy.iterable
+
+import com.daml.lf.speedy.{SExpr, SValue}
+import scala.jdk.CollectionConverters._
+
+object SExprIterable {
```

Comment:
@remyhaemmerle-da I don’t like any of this but I don’t know how to do it nicer. If you have suggestions, I’m very open.

Reply:
@cocreature Maybe add a comment explaining how this iterates. As I understand it, it iterates only over the top layer but doesn't recurse into depth.

Reply:
added a comment, yes your observation is exactly right. Works like our other iterables (e.g. `ExprIterable`) or Haskell’s plated stuff